web-jehlomat/src/routes.ts

```diff
@@ -54,17 +56,24 @@ interface Route {
     protectedRoute?: boolean;
     exact?: boolean;
     title?: string;
+    /*
+     * In case of protectedRoute, remembers previous URL
+     * and redirects to it after successfull login
+     * Valid if protectedRoute is truthy
+     */
+    from?: boolean;
```

Comment:
I like the idea and thinking about how to extend it basically to every protected rote; I mean when the user session expires the natural expectation is to continue where the user left off (I guess we would need to add a new intercepted catching 401 and redirect to the login page with `from` query param). But I guess, we can improve it in the future :)